apps/addons/buttons.py

```diff
@@ -90,7 +90,11 @@ def __init__(self, addon, app, lang, version=None, show_contrib=True,
                  detailed=False, impala=False):
         self.addon, self.app, self.lang = addon, app, lang
         self.latest = version is None
-        self.version = version or addon.current_version
+        self.latest_beta = version == 'beta'
+        if version == 'beta':
+            self.version = addon.current_beta_version
+        else:
+            self.version = version or addon.current_version
```

Comment:
I had a hunch that you wouldn't like it :-)

Heh, will have to do this properly I guess - will make the changes a bit more complicated again.